### DISCUSSION TITLE
What is the command to compile with new make file?

### DISCUSSION THREAD

**Discussion by NeverGray** [Open-SES/OpenSES#17] [Q&A] [answered]
What is the commands to compile the source code using Gfortran or Gmake in Windows Command Prompt or Powershell? 

Previously, I compiled the source code with the makefile using "gmake SES41_GF;". I tried "gmake OpenSES" with the newer makefile, but it gives errors.

**Comment by iamchermac** [comment]
Try 'gmake clean' then 'gmake remake'.

Typing 'gmake' alone should bring up a list of available options.

I had a quick look at the makefile to suggest the tries above.

**Comment by NeverGray** [comment, marked as answer]
Thanks, Chermac. This gave me the 'clue' to figure it out.
If you've compiled before, you should use 'gmake all'.  Afterwards, you can use the different commands of gmake to compile as needed.

```
Available Options:
  clean:        deletes all the object and exe files in release and debug folders
  remake:       deletes obj and exe files then builds all/everything again
  prep:     creates debug and relase folders if they dont exist
  debug:    makes the debug exe file only (if needed)
  release:  makes the release exe file only (if needed)
  all:      makes the debug and release exe files (if needed)
```

**Comment by Yinan-Scott-Shi** [comment]
Hi @NeverGray,

As discussed at the OPENSES in person meeting on Sep 30th, 2022 in Brighton,  here are some tips that we have discussed in email communication.  Hopefully this will help the rest of the community:

1. Install Github Desktop https://desktop.github.com/
2. Install Visual Studio Code  https://code.visualstudio.com/
3. Install Strawberry Perl (this contains the GFortran compiler)  https://strawberryperl.com/

Chapters that are helpful to read in ProGit to start:
1. Chapter 6.2: https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project
2. It's also necessary to read about branching I feel: https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell

Looking forward to see your video on compiling to help first time users.  Please add anything I've missed.
